optuna/visualization/_edf.py

```diff
@@ -22,6 +23,12 @@
 _logger = get_logger(__name__)
 
 
+class _EDFInfo(NamedTuple):
+    study_names: List[str]
+    x_values: np.ndarray
+    y_values_by_study: List[np.ndarray]
```

Comment:
It is implicitly assumed that each member of this `_EDFInfo` has a corresponding name of `study_names` and each element of `y_values_by_study`. To make this explicit, how about the following?

```suggestion
class _EDFLineInfo(NamedTuple):
    study_name: str
    y_values_by_study: np.ndarray


class _EDFInfo(NamedTuple):
    lines: List[_EDFLineInfo]
    x_values: np.ndarray
```